### PR TITLE
[fix](json function) fix failed when json_exists_path use not null input

### PR DESCRIPTION
**Pull request by Mryange**
## Proposed changes
BE has not implemented the code for "return type not nullable."
```
CREATE TABLE test_json (
  id INT not null,
  j JSON not null
)
DUPLICATE KEY(id)
DISTRIBUTED BY HASH(id) BUCKETS 10
PROPERTIES("replication_num" = "1");


INSERT INTO test_json VALUES(26, '{"k1":"v1", "k2": 200}');

SELECT id, j, json_exists_path(j, '$.k1') FROM test_json ORDER BY id;

ERROR 1105 (HY000): errCode = 2, detailMessage = (10.16.10.6)[INTERNAL_ERROR]Function jsonb_exists_path get failed, expr is VectorizedFnCall[jsonb_exists_path](arguments=j, String,return=UInt8) and return type is UInt8. 
```

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Mryange** [comment]
run buildall

**Review by eldenmoon — DISMISSED** [review]
LGTM

**Comment by HappenLee** [comment]
The code seems ugly, we better impl `json_exists_path` in other function not in `json_extract`

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run cloud_p1

File: be/src/vec/functions/function_jsonb.cpp [hunk 1]
```diff
@@ -432,6 +431,151 @@ class FunctionJsonbExtract : public IFunction {
...
+        // doc is NOT necessary to be deleted since JsonbDocument will not allocate memory
+        JsonbDocument* doc = JsonbDocument::createDocument(l_raw_str, l_str_size);
+        if (UNLIKELY(!doc || !doc->getValue())) {
+            return;
+        }
+
+        // value is NOT necessary to be deleted since JsonbValue will not allocate memory
+        JsonbValue* value = doc->getValue()->findValue(path, nullptr);
+
+        if (UNLIKELY(!value)) {
```
Comment (HappenLee) [inline be/src/vec/functions/function_jsonb.cpp:508]: why the is UINLIKELY, seems no need, check origin code.

File: be/src/vec/functions/function_jsonb.cpp [hunk 2]
```diff
@@ -432,6 +431,151 @@ class FunctionJsonbExtract : public IFunction {
...
 
+class FunctionJsonbExtractPath : public IFunction {
+public:
+    static constexpr auto name = "json_exists_path";
+    static constexpr auto alias = "jsonb_exists_path";
+    using ColumnType = ColumnVector<uint8_t>;
+    using Container = typename ColumnType::Container;
+    static FunctionPtr create() { return std::make_shared<FunctionJsonbExtractPath>(); }
+    String get_name() const override { return name; }
+    bool is_variadic() const override { return true; }
```
Comment (HappenLee) [inline be/src/vec/functions/function_jsonb.cpp:442]: shoud not override the func, `get_number_of_arguments` is two

File: be/src/vec/functions/function_jsonb.cpp [hunk 3]
```diff
@@ -432,6 +431,151 @@ class FunctionJsonbExtract : public IFunction {
...
+    size_t get_number_of_arguments() const override { return 2; }
+    DataTypePtr get_return_type_impl(const DataTypes& arguments) const override {
+        // it only needs to indicate existence and does not need to return nullable values.
+        return std::make_shared<DataTypeUInt8>();
+    }
+
+    Status execute_impl(FunctionContext* context, Block& block, const ColumnNumbers& arguments,
+                        size_t result, size_t input_rows_count) const override {
+        DCHECK_EQ(arguments.size(), 2);
```
Comment (HappenLee) [inline be/src/vec/functions/function_jsonb.cpp:451]: remove the linecode, if function `is_variadic() ` false will do the chcek before exec

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by Gabriel39 — APPROVED** [review]

**Pull request merged**